pkg/db/src/types/orders.ts

```diff
@@ -9,10 +9,55 @@ export interface NewOrdersDatabase {
 }
 
 /**
- * A (standardized) customer order interface:
+ * Standardized supplier interface
+ */
+
+export type SupplierInterface<A extends Record<string, any> = {}> = SupplierProto<A> & SupplierData<A>;
+
+/**
+ * Methods interface
+ */
+export interface SupplierProto<A extends Record<string, any> = {}> {
+	create: () => Promise<SupplierInterface<A>>;
+}
+
+/** Data structure */
+export type SupplierData<A extends Record<string, any> = {}> = CouchDocument<
+	{
+		updatedAt: string | null;
+		publishers: string[];
+		displayName: string;
+	} & A
+>;
+/**
+ * Standardized supplier order interface
+ */
+
+export type SupplierOrderInterface<A extends Record<string, any> = {}> = SupplierOrderProto<A> & SupplierOrderData<A>;
+
+/**
+ * Methods interface
+ */
+export interface SupplierOrderProto<A extends Record<string, any> = {}> {
+	create: () => Promise<SupplierOrderInterface<A>>;
+}
```

Comment:
The proto should contain methods:
- create
- delete
- commit*
- csv - this should be available only on committed batches
- addItems - akin to `addTransactions` in the note interface
- stream:
  - displayName
  - items
  - state
  - updatedAt
 

*Let's use `commit` instead of `order` as the latter might imply:
- order interface (one that doesn't exist, but would be consistent with our api, something like `db.supplier("supplier-1").batch("order-batch-1").order("order-1")`)
- `order` - as in `sort`
- `order` - the actual use case (place the order)
Using `commit` removes ambiguity and keeps consistent with the rest of the db interface